### PR TITLE
Add queries custom and test

### PR DESCRIPTION
**Pull request by luismiguel0529**
## Summary
- Add methods complex with anotation @Query in repositories
- Add test of method complex
## Trello Card
https://trello.com/c/CiPnxabX/30-queries-custom-en-repositorio

### CODE REVIEW

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 1]
```diff
@@ -21,4 +22,12 @@
 
     public Optional<List<Book>> findByPublisherAndGenreAndYear(String publisher, String genre, String year);
 
+    @Query("SELECT b from Book b "
```
Comment (joredudiaz) [inline src/main/java/wolox/training/repositories/BookRepository.java:54]: This is good, can you add it documentation?, please

File: src/main/java/wolox/training/repositories/UsersRepository.java [hunk 2]
```diff
@@ -18,7 +19,16 @@
 
     public Optional<User> findByUsername(String username);
 
-    List<User> findAllByBirthdateBetweenAndNameContainingIgnoreCase(
+    Optional<List<User>> findAllByBirthdateBetweenAndNameContainingIgnoreCase(
```
Comment (joredudiaz) [inline src/main/java/wolox/training/repositories/UsersRepository.java]: This is good, can you add it documentation?, please

File: src/main/java/wolox/training/repositories/UsersRepository.java [hunk 3]
```diff
@@ -18,7 +19,16 @@
...
+            LocalDate startDate,
+            LocalDate endDate,
+            String name);
+
+    @Query("SELECT u FROM User u"
```
Comment (joredudiaz) [inline src/main/java/wolox/training/repositories/UsersRepository.java:50]: This is good, can you add it documentation?, please

File: src/test/java/wolox/training/repositories/BookRepositoryTest.java [hunk 4]
```diff
@@ -54,4 +54,18 @@ void whenSaveBookThenJpaPersisted() {
...
+
+    @Test
+    void whenCallfindByPublisherAndGenreAndYearThenReturnListBook() {
+        bookRepository.save(oneTestBook);
+        Optional<List<Book>> books = bookRepository.findByPublisherAndGenreAndYear(oneTestBook.getPublisher(), oneTestBook.getGenre(), oneTestBook.getYear());
+        assertEquals(books.get().get(0).getPublisher(), oneTestBook.getPublisher());
+    }
+
+    @Test
+    void whenCallfindByPublisherAndGenreAndYearQueryThenReturnListBook() {
```
Comment (joredudiaz) [inline src/test/java/wolox/training/repositories/BookRepositoryTest.java:66]: you also should test it with null parameters

File: src/test/java/wolox/training/repositories/UserRepositoryTest.java [hunk 5]
```diff
@@ -48,4 +47,22 @@ void whenUpdateUserThenReturnUserUpdated() {
...
+    void whenCallFindAllByBirthdateBetweenAndNameContainingIgnoreCaseThenReturnListUser() {
+        LocalDate startDate = LocalDate.of(1992, 11, 11);
+        LocalDate endDate = LocalDate.of(2020, 11, 11);
+        usersRepository.save(oneTestUser);
+        Optional<List<User>> users = usersRepository.findAllByBirthdateBetweenAndNameContainingIgnoreCase(startDate, endDate, oneTestUser.getName());
+        assertEquals(users.get().get(0).getBirthdate(), oneTestUser.getBirthdate());
+    }
+
+    @Test
+    void whenCallFindAllByBirthdateBetweenAndNameContainingIgnoreCaseQueryThenReturnListUser() {
```
Comment (joredudiaz) [inline src/test/java/wolox/training/repositories/UserRepositoryTest.java]: you also should test it with null parameters

**Review by joredudiaz — CHANGES_REQUESTED** [review]
😄 a couple comments

File: src/main/java/wolox/training/repositories/BookRepository.java (hunk 1 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/repositories/BookRepository.java:54]: corrected

File: src/main/java/wolox/training/repositories/UsersRepository.java (hunk 2 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/repositories/UsersRepository.java]: corrected

File: src/main/java/wolox/training/repositories/UsersRepository.java (hunk 3 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/repositories/UsersRepository.java:50]: corrected

File: src/test/java/wolox/training/repositories/BookRepositoryTest.java (hunk 4 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/repositories/BookRepositoryTest.java:66]: corrected

**Review by joredudiaz — DISMISSED** [review]
I look at it good 😄

**Review by joredudiaz — APPROVED** [review]

**Review by facucompe — APPROVED** [review]
